pygments/lexers/esoteric.py

```diff
@@ -48,6 +48,15 @@ class BrainfuckLexer(RegexLexer):
         ]
     }
 
+    def analyse_text(text):
+        """It's safe to assume that any reasonable brainfuck program will have
+        more than two consecutive + signs, which is very uncommon in any
+        other programming language."""
+        if '+++++' in text or '------' in text:
+            return 0.5
+        if '+++' in text:
```

Comment:
`+++` and `---` are common in diff/patch files, which are probably more common than brainfuck :)

The longer `+++++++` and `------` sequences are typical for underlines in readable markup like rst/markdown.

Another signature of brainfuck which has less conflicts is probably `[-]`, i.e. "clear this cell".

Reply:
I've changed it to check if 25% of the source consist of +- or <>, that seems fairly unlikely in any other source (even HTML is not that <> heavy.)

Reply:
While this is probably a good test, I'm worrying about the performance implications of looping over the string on the Python level... the other tests at least use the C-based regex/stringlib routines.

Reply:
Given it's statistics, maybe trying it on the first 256 characters or so is enough? The character distribution for brainfuck should be similar across any subset of the program.